test/gtest/ucs/test_rcache.cc

```diff
@@ -828,3 +829,77 @@ UCS_TEST_F(test_rcache_stats, hits_slow) {
     munmap(mem2, size1);
 }
 #endif
+
+
+class test_rcache_pfn : public ucs::test {
+public:
+    void test_pfn(void *address, unsigned page_num)
+    {
+        pfn_enum_t ctx;
+        ucs_status_t status;
+
+        ctx.page_num = page_num;
+        status       = ucs_sys_enum_pfn((uintptr_t)address,
+                                        page_num, enum_pfn_cb, &ctx);
+        ASSERT_UCS_OK(status);
+        ASSERT_TRUE(ctx.page.size() == page_num);
+        ASSERT_TRUE(ctx.pfn.size() == page_num);
+    }
+
+protected:
+    typedef std::set<unsigned> page_set_t;
+    typedef std::set<unsigned long> pfn_set_t;
+    typedef struct {
+        unsigned   page_num;
+        page_set_t page;
+        pfn_set_t  pfn;
+    } pfn_enum_t;
+
+    static void enum_pfn_cb(unsigned page_num, unsigned long pfn, void *ctx)
+    {
+        pfn_enum_t *data = (pfn_enum_t*)ctx;
+
+        EXPECT_LT(page_num, data->page_num);
+        EXPECT_EQ(data->pfn.find(pfn), data->pfn.end());
+        EXPECT_EQ(data->page.find(page_num), data->page.end());
+        data->pfn.insert(pfn);
+        data->page.insert(page_num);
+    }
+};
+
+UCS_TEST_F(test_rcache_pfn, enum_pfn) {
+    const int MAX_PAGE_NUM = 1024 * 100; /* 400Mb max buffer */
+    size_t page_size       = ucs_get_page_size();
+    void *region;
+    unsigned i;
+    size_t len;
+    unsigned long pfn;
+    ucs_status_t status;
+
+    /* check if PFN is supported */
+    status = ucs_sys_get_pfn((uintptr_t)&pfn, 1, &pfn);
+    ASSERT_UCS_OK(status);
+    if (pfn == 0) {
+        /* stack page could not be mapped into zero region */
```

Comment:
` /* stack page could not be mapped into zero region, if we get 0 here it means the kernel does not provide PFNs */`

Reply:
fixed